rust/entity/src/job.rs

```diff
@@ -9,7 +9,8 @@ pub struct Model {
     pub id: Uuid,
     #[sea_orm(column_type = "Binary(BlobSize::Blob(None))", unique)]
     pub hash: Vec<u8>,
-    pub cmd: String,
+    #[sea_orm(column_type = "Binary(BlobSize::Blob(None))")]
+    pub cmd: Vec<u8>,
```

Comment:
Can `cmd`'s contain unicode? I assume rust's `String` could but not sure if we actually used that? Not enough of a unicode person to know whether a vec of bytes is an appropriate representation but wanted to double check, since it seems potentially problematic.

Reply:
> Can `cmd`'s contain unicode?

Not sure but we should assume yes since wake strings can

> since it seems potentially problematic

We have to use a Vec<u8> here because we stuff null bytes into the `cmd` string where they don't belong. Which causes Rust's string type to (correctly) fail. (Can't remember if it panics or just fails to parse). There is no harm in treating unicode strings (even if they are multi-byte code points) as a vector of bytes as long as we never try to use that vector as a string later. In reality all we do with this vec of bytes is hash & store it for matching. We don't even return it to the client on a job match because the client already know the cmd in that case

Reply:
Gotcha since it just becomes data that makes sense. I guess we just want to make sure people looking at the DB don't expect to be able to copy the command out of this field, but hopefully that is an easy thing to do given the number of people looking at the DB :)